internal/pkg/manifest/http.go

```diff
@@ -97,6 +97,22 @@ func (r *RoutingRule) IsEmpty() bool {
 		r.HostedZone == nil && r.RedirectToHTTPS == nil
 }
 
+// HealthCheckPort returns the port a HealthCheck is set to for a RoutingRule.
+func (r *RoutingRule) HealthCheckPort(mainContainerPort *uint16) uint16 {
+	// healthCheckPort is defined by RoutingRule.HealthCheck.Port, with fallback on RoutingRule.TargetPort, then image.port.
+	var healthCheckPort uint16
+	if mainContainerPort != nil {
+		healthCheckPort = aws.Uint16Value(mainContainerPort)
+	}
+	if r.TargetPort != nil {
+		healthCheckPort = aws.Uint16Value(r.TargetPort)
+	}
+	if r.HealthCheck.Advanced.Port != nil {
+		healthCheckPort = uint16(aws.IntValue(r.HealthCheck.Advanced.Port))
+	}
+	return healthCheckPort
+}
```

Comment:
nit: i wonder if the logic is clearer if the order is reversed, like:
```suggestion
	// healthCheckPort is defined by RoutingRule.HealthCheck.Port, with fallback on RoutingRule.TargetPort, then image.port.
	if r.HealthCheck.Advanced.Port != nil {
		return uint16(aws.IntValue(r.HealthCheck.Advanced.Port))
	}
	if r.TargetPort != nil {
		return aws.Uint16Value(r.TargetPort)
	}
	if mainContainerPort != nil {
		return aws.Uint16Value(mainContainerPort)
	}
	return 0
}
```

Reply:
I like this 👍